aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/DeploymentSteps.java

```diff
@@ -139,6 +143,75 @@ SdkIterable<GroupNameAndArn> getThingGroupIterable(String coreThingName) {
         return lifecycle.listThingGroupsForAThing(coreThingName).thingGroups();
     }
 
+    /**
+     * implemented the step of installing a custom component with configuration. This requires Recipe for the custom
+     * compoenent.
+     *
+     * @param componentName         the name of the custom component
+     * @param configurationTable    the table which describes the configurations
+     * @throws InterruptedException InterruptedException could be throw out during the component deployment
+     * @throws IOException          IOException could be throw out during preparation of the CLI command
+     */
+    @When("I install the component {word} from local store with configuration otf")
+    public void installComponentWithConfiguration(final String componentName, final String configurationTable)
+            throws InterruptedException, IOException {
+        // read the recipe from local store, and get component name and version from recipe
+        List<String> componentSpecs = Arrays.asList(
+                componentName, LOCAL_STORE_RECIPES.resolve(String.format("%s.yaml", componentName)).toString()
+        );
+
+        // handle the config input
+        List<Map<String, Object>> configuration = readConfigurationTable(configurationTable);
+        createLocalDeploymentWithConfig(new ArrayList<>(Collections.singleton(componentSpecs)),
+                configuration, 0);
+    }
+
+    private List<Map<String, Object>> readConfigurationTable(String configurationTable) throws JsonProcessingException {
+        List<Map<String, Object>> configuration = new ArrayList<>();
+        String updatedConfiguration = this.scenarioContext.applyInline(configurationTable);
+        Map<String, Object> json = this.mapper.readValue(updatedConfiguration,
+                new TypeReference<Map<String, Object>>() {});
+        configuration.add(json);
+        return configuration;
+    }
+
+    private void createLocalDeploymentWithConfig(List<List<String>> componentNames,
+                                                 List<Map<String, Object>> configuration,
+                                                 int retryCount) throws InterruptedException, IOException {
+        // find the component artifacts and copy into a local store
+        final Map<String, ComponentDeploymentSpecification> components = parseComponentNamesAndPrepare(componentNames);
+
+        List<String> commandArgs = new ArrayList<>();
+
+        commandArgs.addAll(Arrays.asList("deployment", "create",
+                "--artifactDir "  + artifactPath.toString(),
+                "--recipeDir " + recipePath.toString()));
+
+        for (Map.Entry<String, ComponentDeploymentSpecification> entry : components.entrySet()) {
+            commandArgs.add(" --merge ");
+            String componentName = entry.getKey();
+            commandArgs.add(componentName + "=" + entry.getValue().componentVersion());
```

Comment:
Shall we move --merge in line to this line as well? And I think it is better to use `String.format` here instead of +?